offline_compiler/utilities/linux/safety_guard_linux.h

```diff
@@ -28,6 +30,7 @@ class SafetyGuardLinux {
     }
 
     static void sigAction(int sig_num, siginfo_t *info, void *ucontext) {
+#if !defined(__ANDROID__)
```

Comment:
please don't put code under `if defined` . If android needs other implementation, it should be placed in separated file, i.e. one file for linux, one file for android - cmake chooses correct file based on `ANDROID` flag.

Reply:
What about `drm_neo.cpp`? There are about 40 files there.

Reply:
If Linux version of any class doesn't meet Android requirements, then you should create Android specific implementation in separated file. Currently we have such separation between Linux and Windows.

Please see:
`runtime\dll\linux\create_drm_memory_manager.cpp`
`runtime\dll\windows\create_wddm_memory_manager.cpp`
They both implement single method `std::unique_ptr<MemoryManager> MemoryManager::createMemoryManager(ExecutionEnvironment &executionEnvironment)`